src-tauri/src/telemetry_manager.rs

```diff
@@ -408,3 +422,40 @@ async fn send_telemetry_data(
     }
     Ok(None)
 }
+
+async fn retry_with_backoff<T, R, E>(
+    mut f: T,
+    increment_in_secs: usize,
+    max_retries: u64,
+    operation_name: &str,
+) -> anyhow::Result<R>
+where
+    T: FnMut() -> Pin<Box<dyn Future<Output = Result<R, E>> + Send>>,
+    E: std::error::Error,
+{
+    let range_size = increment_in_secs * max_retries as usize + 1;
+
+    for i in (0..range_size).step_by(increment_in_secs) {
+        tokio::time::sleep(Duration::from_secs(i as u64)).await;
```

Comment:
We only do 3s and 6s backoff I think this is not too much for now. 
Also jitter is not needed as our users are starting up there applications in random times so that gives enough jitter altogether.